lib/membrane_audio_mixer_bin.ex

```diff
@@ -90,123 +63,123 @@ defmodule Membrane.AudioMixerBin do
     availability: :always,
     caps: Raw
 
+  defmodule MixerOptions do
+    @moduledoc """
+    Structure representing options that would be passed to each created Membrane.AudioMixer element.
+    """
+    defstruct caps: nil, frames_per_buffer: 2048, prevent_clipping: true
+
+    @type t :: %__MODULE__{
+            caps: Raw.t(),
+            frames_per_buffer: pos_integer(),
+            prevent_clipping: boolean()
+          }
+  end
+
   @impl true
   def handle_init(options) do
     state =
       options
       |> Map.from_struct()
-      |> Map.put(:added_pad_count, 0)
-
-    {children, links} = create_mixers_tree(state)
-    spec = %ParentSpec{children: children, links: links}
+      |> Map.put(:inputs, 0)
+      |> Map.update!(:mixer_options, fn val ->
+        if val == nil, do: %MixerOptions{}, else: val
+      end)
 
-    {{:ok, spec: spec}, state}
+    {{:ok, spec: %ParentSpec{}}, state}
   end
 
   @impl true
-  def handle_pad_added(
-        _pad_ref,
-        _ctx,
-        %{added_pad_count: added_pad_count, inputs: inputs}
-      )
-      when added_pad_count >= inputs do
-    raise("provided more input pads than specified via `inputs` option (#{inputs})")
-  end
-
-  def handle_pad_added(
-        pad_ref,
-        ctx,
-        %{added_pad_count: added_pad_count, max_node_degree: max_node_degree} = state
-      ) do
+  def handle_pad_added(pad_ref, %{playback_state: :stopped} = ctx, %{inputs: inputs} = state) do
     %Pad.Data{options: %{offset: offset}} = ctx.pads[pad_ref]
-    mixer_idx = div(added_pad_count, max_node_degree)
+    {children, links} = link_new_input(pad_ref, offset, state)
 
-    link =
-      link_bin_input(pad_ref)
-      |> via_in(:input, options: [offset: offset])
-      |> to("mixer_0_#{mixer_idx}")
+    {{:ok, spec: %ParentSpec{children: children, links: links}}, %{state | inputs: inputs + 1}}
+  end
 
-    {{:ok, spec: %ParentSpec{links: [link]}}, %{state | added_pad_count: added_pad_count + 1}}
+  def handle_pad_added(_pad_ref, %{playback_state: playback_state}, _state)
+      when playback_state != :stopped do
+    raise("All pads should be added before starting the #{__MODULE__}. \
+Pad added event received in playback state.")
```

Comment:
1. You can make it multi-line

```suggestion
    raise("""
    All pads should be added before starting the #{__MODULE__}.
    Pad added event received in playback state.
    """)
```
2. The second sentence seems to be missing something - did you forget to inspect playback state?